scripts/zones/Arrapago_Reef/mobs/Medusa.lua

```diff
@@ -1,28 +1,45 @@
 -----------------------------------
 -- Area: Arrapago Reef
 -- NPC:  Medusa
+-- TODO: resists, attack/def boosts
 -----------------------------------
 
 require("scripts/globals/titles");
+require("scripts/zones/Arrapago_Reef/MobIDs");
+require("scripts/zones/Arrapago_Reef/TextIDs");
 
 -----------------------------------
 -- onMobSpawn Action
 -----------------------------------
 
 function onMobSpawn(mob)
+    mob:setLocalVar("eeshpp", math.random(5,99)); -- Uses EES randomly during the fight
 end;
 
 -----------------------------------
 -- onMobEngaged Action
 -----------------------------------
 
-function onMobEngaged(mob,target)
+function onMobEngaged(mob, target)
+    target:showText(mob, MEDUSA_ENGAGE);
+    SpawnMob(ExonOne,   180):updateEnmity(target);
+    SpawnMob(ExonTwo,   180):updateEnmity(target);
+    SpawnMob(ExonThree, 180):updateEnmity(target);
+    SpawnMob(ExonFour,  180):updateEnmity(target);
+end;
 
-	SpawnMob(16998863,180):updateEnmity(target);
-	SpawnMob(16998864,180):updateEnmity(target);
-	SpawnMob(16998865,180):updateEnmity(target);
-	SpawnMob(16998866,180):updateEnmity(target);
+-----------------------------------
+-- onMobFight Action
+-----------------------------------
 
+function onMobFight(mob, target)
+    local HPP = mob:getHPP();
+    if (mob:getLocalVar("usedees") == 0) then
+        if (HPP <= mob:getLocalVar("eeshpp")) then
+            mob:useMobAbility(479); -- Eagle Eye Shot

```

Comment:
@bendangelo 

> if(mobutils.hasUsedTwoHour(mob, JOB_THF) == false) then
mobutils.useTwoHour(mob, JOB_THF)
end

You just repeated my suggestion and called it cleaner. 

Except I think it saner to say the 2hr name, not the job abbreviation - I do **not** want a bunch of conditionals figuring out which EES to call, lets just call the correct one in the first place its not that hard for scripters to test and see if it animates correct. You are saying the using the input is hard coding, which is precisely the opposite..By setting the ID base don a non input criteria you are hard coding the result. Doing so is not cleaner, its using hard coding to solve what you see as hard coding. 

For the record, I detest whats in core - why would you tie what 2hr they can use to their job at all? Many mobs use a 2hr that doesn't match their job with the most notable being AV and the ark angels. Its the worst possible way to handle them! 

>what if we discover, some animations were wrong? Now we have to fix multiple scripts instead of fixing one method

The animations aren't even part of the script. What we would do is add the alternates to the table with a different animation ID, and call that ID instead. Not exactly an earth shattering bug there.


Btw we should probably avoid saying TwoHour or TwoHr etc btw, since they aren't even called that in retail anymore. Not on 2 hour timers any more (and in DSP mobs generally get them once per spawn which is very likely retail behavior) so calling it that may cause some noob confusion in some distant future despite our familiarity with the term right now.

@nesstea I would hazard a guess that what retail does for these is the chance to "go off" increases as the HP decreases. rather than just being a random number. This weights it toward lower HP values while still letting it happens at a wide range of values. However, me trying to prove that would make my head asplode.